### PR TITLE
improve Airflow integration docs

### LINKED ISSUE #3891: [Content Gap] Add documentation for "Airflow Inject"
## What content is missing

This is actually our most well-supported codepath in the airflow and not mentioned at all in airflow.mdx. We should also be clear that the "ingest" is pretty experimental

## Writer's Guide
- [Docs README](https://github.com/dagster-io/dagster/tree/master/docs)
- Templates to follow: <fill by the docs team>
- Examples for reference: <fill by the docs team>

---
#### Message from the maintainers:
Are you looking for the same documentation content? Give it a :thumbsup:. We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by sryza**
The Airflow integration docs were confusingly spread across two different places, which made many readers believe that we either only supported Airflow ingestion or Airflow export not both.

They were originally part of the intro tutorial as far as I can tell, which leads to the main points kind of getting lost in a sea of explaining Docker and explaining Airflow.  That content made sense when Airflow was the recommended way of deploying Dagster, but makes less sense where the main scenario for coming to the integration is already having an existing functioning Airflow setup.  So this version cuts that stuff out.

Fixes #3891

### CODE REVIEW

File: docs/content/deployment/guides/airflow.mdx [hunk 1]
```diff
@@ -70,54 +70,7 @@ Wrote DAG scaffold to file: $AIRFLOW_HOME/dags/hello_cereal_pipeline.py
...
-
-dag, tasks = make_airflow_dag(
-    # NOTE: you must ensure that docs_snippets.intro_tutorial.airflow is
-    # installed or available on sys.path, otherwise, this import will fail.
-    module_name="docs_snippets.intro_tutorial.airflow",
-    pipeline_name="hello_cereal_pipeline",
-    run_config=yaml.safe_load(ENVIRONMENT),
-    dag_kwargs={"default_args": DEFAULT_ARGS, "max_active_runs": 1},
-)
+ENOENT: no such file or directory, open '/Users/sryza/dagster/examples/docs_snippets/docs_snippets/deploying/airflow/generated_airflow.py'
```
Comment (yuhan) [inline docs/content/deployment/guides/airflow.mdx:73]: [1]

Comment (yuhan) [inline docs/content/deployment/guides/airflow.mdx:73]: if we still this content, we need to update the filepath.

File: docs/content/deployment/guides/airflow.mdx [hunk 2]
```diff
@@ -247,18 +200,7 @@ For most production applications, you'll probably want to use a private Docker r
...
-    module_name="docs_snippets.intro_tutorial.airflow",
-    pipeline_name="hello_cereal_pipeline",
-    image="dagster-airflow-demo-repository",
-    run_config={"storage": {"filesystem": {"config": {"base_dir": "/tmp"}}}},
-    dag_id=None,
-    dag_description=None,
-    dag_kwargs=None,
-    op_kwargs=None,
-)
+ENOENT: no such file or directory, open '/Users/sryza/dagster/examples/docs_snippets/docs_snippets/deploying/airflow/containerized.py'
```
Comment (yuhan) [inline docs/content/deployment/guides/airflow.mdx:203]: filepath too

File: docs/content/deployment.mdx [hunk 3]
```diff
@@ -42,4 +42,3 @@ This section provides a series of hands-on guides to walk you through Dagster de
 | [Deploying Dagster to GCP](/deployment/guides/gcp)         | This guide explains how to install and deploy Dagster to GCP GCE.                        |
 | [Executing on Celery](/deployment/guides/celery)           | This guide explains how to execute Dagster on Celery.                                    |
 | [Executing on Dask](/deployment/guides/dask)               | This guide explains how to execute Dagster on Dask.                                      |
-| [Executing on Airflow](/deployment/guides/airflow)         | This guide explains how to execute Dagster on Airflow.                                   |
```
Comment (yuhan) [inline docs/content/deployment.mdx:45]: is the plan to remove this guide completely? if yes, we should remove `docs/content/deployment/guides/airflow.mdx` too?

File: docs/content/integrations/airflow.mdx [hunk 4]
```diff
@@ -1,18 +1,135 @@
...
+    sorted_by_sugar = sorted(cereals, key=lambda cereal: cereal["sugars"])
+    context.log.info(f'{sorted_by_sugar[-1]["name"]} is the sugariest cereal')
+
+
+@pipeline(mode_defs=[ModeDefinition(resource_defs={"io_manager": fs_io_manager})])
+def hello_cereal_pipeline():
+    find_sugariest(download_cereals())
+```
+
+Notice that this pipeline uses a mode with the <PyObject module="dagster" object="fs_io_manager" />. This is important because each Airflow task executes in its own process. Therefore, we need to write outputs to the filesystem so that downstream tasks can access them, instead of fetching them from memory, which is Dagster's default.
```
Comment (yuhan) [inline docs/content/integrations/airflow.mdx:46]: the filesystem wording may confuse people who are using the remote set up. we could rephrase it to something like using persistent io managers in order to pass data between processes. and also mention s3_io_manager, gcs_io_manager for remote options.

Comment (sryza) [reply docs/content/integrations/airflow.mdx]: Good point - I'll change

**Review by yuhan — APPROVED** [review]
looking great!
